### ISSUE TITLE
How to change name servers of a domain purchased with now-cli

### ISSUE THREAD

**Issue by olivierlesnicki** [vercel/vercel#618]
How to change name servers of a domain purchased with now-cli?

I know I can use `now dns` to update the `dns` records. But how do I repoint the `nameservers`? (I want to point them to Cloudflare for example).

**Comment by rauchg** [comment]
This is currently planned and will be implemented soon. For now, you can take advantage of our free DNS, which supports all record types!

**Comment by rauchg** [comment]
Also, if needed, we can manually process a change. Please email `support@zeit.co`. Apologize for the inconvenience!

**Comment by olivierlesnicki** [comment]
There's no inconvenience whatsoever! I think what you guys are doing is fantastic, I'm currently on a  Premium plan!

I have an Heroku application and I've used my domain bought with now. Everything works fine except https. See for yourself: https://api.theviemusic.com/tracks.

Cloudflare takes care of generating valid terminating SSL quite easily. Hence my question.

**Comment by leo** [comment]
Thank you! ❤️ As this is already planned, I'm going to close this issue.

**Comment by abattenburg** [comment]
@leo Was there ever an update to this issue? I'm not able to find any documentation on managing domains bought through now-cli beside the cli itself. Thanks for pointing me in the right direction!

**Comment by stephenfjohnson** [comment]
Is there a timeline to change the nameservers in the cli?

**Comment by cs-miller** [comment]
+1 on this.  I would like to change nameservers to Netlify DNS.

**Comment by nodegin** [comment]
Still waiting on this!

**Comment by danrasmuson** [comment]
@leo how is this accomplished?

**Comment by alexhoma** [comment]
Any updates about this? 😄

**Comment by LuisPaGarcia** [comment]
Any update?

**Comment by josuemotte** [comment]
@rauchg , does it mean that for now we won't be able to use the external command correctly ? And that we will have to contact support@zeit.co each time ?

**Comment by michaelcassius** [comment]
@josuemotte pretty much yes. at least Zeit are very fast responding and doing this - **but it's beyond me why they're doing it this way at all** , can't see any benefit it's just more manual work for them and delays for us ... hopefully this changes soon.

**Comment by jamo** [comment]
Apologies for the inconvenience caused; this feature was originally postponed because of other DNS improvements; including automatic DNS records.

I'm happy to announce that this issue is finally being worked at and available in both CLI and Web very soon.

**Comment by ForsakenHarmony** [comment]
👀

**Comment by parhelium** [comment]
Any progress on that ?

**Comment by jamo** [comment]
This feature is still on our roadmap but has been delayed because of a couple of other domains/dns improvements - stay tuned to our blog zeit.co/blog

We are consolidating all feature requests to https://spectrum.chat/zeit/feature-requests to better track these.

**Comment by karlhorky** [comment]
@jamo do you have a link to this feature request on Spectrum?

I suppose the process as of December 2018 to change nameservers for domains bought over the command line and also directly on the website at https://zeit.co/domains is still to email support@zeit.co ?

Edit: Yes, it seems like this is the current process 😬

**Comment by 0xbkt** [comment]
This feature should come asap.

**Comment by simonhaenisch** [comment]
@karlhorky just opened one: https://spectrum.chat/zeit/feature-requests?thread=fdacc307-5c5c-4497-ace1-d86f1f0e8917.

**Comment by 0x8801** [comment]
Additional request for the time being - Maybe change this message `If you want to change the domain to be external, please remove it and then add it back as an external domain.` to something that communicates that this feature is not available (for an NS change) and the user will need to request manually with support@zeit.co

**Comment by callumlocke** [comment]
Wait, what? If I purchase a domain through Zeit, the only way to change the nameservers away from Zeit is to email the support? Is that actually still the case?

**Comment by simonhaenisch** [comment]
Yup that's how I did it last time (was a while ago), they are pretty quick though. (One of the reasons I stopped buying domains there 🤷🏻‍♂️)

**Comment by equinusocio** [comment]
https://spectrum.chat/zeit/feature-requests/ability-to-change-name-servers~fdacc307-5c5c-4497-ace1-d86f1f0e8917?m=MTU3MTI0MjY2MDA1OQ==

It seems they don't want to add this feature reading the feature request on spectrum. So i have to switch to netlify.

**Comment by callumlocke** [comment]
The reasoning given on that [thread](https://spectrum.chat/zeit/feature-requests/ability-to-change-name-servers~fdacc307-5c5c-4497-ace1-d86f1f0e8917) is worrying:

> "...we are not focused on being a registrar, but a host solution ... Advanced settings that prevents foot-guns, like change of NameServers and other features, should be indeed reserved"

Changing nameservers is not an advanced setting. It's a basic part of managing a domain that I own.

Zeit lets me change MX records without a support email – isn't that a foot-gun too? What's the difference, other than staying within the Zeit ecosystem? When a hosting company makes it possible to self-manage everything _except_ moving a component away from them, it's hard not to be cynical about their motives.

**Comment by elitan** [comment]
Zeit is such an awesome product but here is obviously a gap. Sending an e-mail to update the nameserver? What is this? 1997? :o

**Comment by renslaros** [comment]
3 years later...

**Comment by radiumrasheed** [comment]
3 years 3 days later .... 😒😒

**Comment by hackermondev** [comment]
3 years 10+ days lates

**Comment by radiumrasheed** [comment]
I think they now have a web interface for this

**Comment by hackermondev** [comment]
OwO, does it only work for a domain purchased with now or can you do it with a domain connected to now

**Comment by ForsakenHarmony** [comment]
I don't see any interface, I already configured it through support though, so maybe that would disable the option

But there's nothing greyed out either

Just "Third Party" nameservers
![](https://i.hrmny.sh/hUWpC8NUnp.png)

**Comment by dan-turner** [comment]
Why is this issue closed?

**Comment by mcsdevv** [comment]
This issue is closed because we are managing it internally.

Now that the DNS UI is in place, which was previously holding this back, we will be able to implement this shortly.

**Comment by KameronKales** [comment]
+1 on wanting to know when this is done.
